stacks-node/src/burnchains/rpc/bitcoin_rpc_client/test_utils.rs

```diff
@@ -0,0 +1,183 @@
+    ///   - A raw hex-encoded transaction
+    ///   - A transaction ID (must be present in the mempool)
+    ///   If the list is empty, an empty block (with only the coinbase transaction) will be generated.
+    ///
+    /// # Returns
+    /// The block hash of the newly generated block.
+    ///
+    /// # Availability
+    /// - **Since**: Bitcoin Core **v22.0**.
+    /// - Requires `regtest` or similar testing networks.
+    pub fn generate_block(&self, address: &str, txs: &[&str]) -> BitcoinRpcClientResult<String> {
+        let response = self.global_ep.send::<GenerateBlockResponse>(
+            &self.client_id,
+            "generateblock",
+            vec![address.into(), txs.into()],
+        )?;
+        Ok(response.hash)
+    }
+
+    /// Gracefully shuts down the Bitcoin Core node.
+    ///
+    /// Sends the shutdown command to safely terminate `bitcoind`. This ensures all state is written
+    /// to disk and the node exits cleanly.
+    ///
+    /// # Returns
+    /// On success, returns the string: `"Bitcoin Core stopping"`
+    ///
+    /// # Availability
+    /// - **Since**: Bitcoin Core **v0.1.0**.
+    pub fn stop(&self) -> BitcoinRpcClientResult<String> {
+        Ok(self.global_ep.send(&self.client_id, "stop", vec![])?)
+    }
+
+    /// Retrieves a new Bitcoin address from the wallet.
+    ///
+    /// # Arguments
+    /// * `label` - Optional label to associate with the address.
+    /// * `address_type` - Optional address type (`"legacy"`, `"p2sh-segwit"`, `"bech32"`, `"bech32m"`).
+    ///   If `None`, the address type defaults to the node’s `-addresstype` setting.
+    ///   If `-addresstype` is also unset, the default is `"bech32"` (since v0.20.0).
+    ///
+    /// # Returns
+    /// A string representing the newly generated Bitcoin address.
+    ///
+    /// # Availability
+    /// - **Since**: Bitcoin Core **v0.1.0**.  
+    /// - `address_type` parameter supported since **v0.17.0**.
+    /// - Defaulting to `bech32` (when unset) introduced in **v0.20.0**.
+    pub fn get_new_address(
+        &self,
+        label: Option<&str>,
+        address_type: Option<&str>,
+    ) -> BitcoinRpcClientResult<String> {
+        let mut params = vec![];
+
+        let label = label.unwrap_or("");
+        params.push(label.into());
+
+        if let Some(at) = address_type {
+            params.push(at.into());
+        }
+
+        Ok(self
+            .global_ep
+            .send(&self.client_id, "getnewaddress", params)?)
+    }
+
+    /// Sends a specified amount of BTC to a given address.
+    ///
+    /// # Arguments
+    /// * `address` - The destination Bitcoin address.
+    /// * `amount` - Amount to send in BTC (not in satoshis).
+    ///
+    /// # Returns
+    /// The transaction ID as hex string
+    ///
+    /// # Availability
+    /// - **Since**: Bitcoin Core **v0.1.0**.
+    pub fn send_to_address(&self, address: &str, amount: f64) -> BitcoinRpcClientResult<String> {
```

Comment:
Please use `BitcoinAddress` instead of `&str`.  Please also use `Txid` instead of `String`.